torchtune/_cli/download.py

```diff
@@ -95,8 +111,37 @@ def _add_arguments(self) -> None:
             help="If provided, files matching any of the patterns are not downloaded. Defaults to ignoring "
             "safetensors files to avoid downloading duplicate weights.",
         )
+        self._parser.add_argument(
+            "--source",
+            type=str,
+            required=False,
+            default="huggingface",
+            choices=["huggingface", "kaggle"],
+            help="If provided, downloads model weights from the provided repo_id on the designated source hub.",
+        )
+        self._parser.add_argument(
+            "--kaggle-username",
+            type=str,
+            required=False,
+            help="Kaggle username for authentication. Needed for gated models like Llama2.",
+        )
+        self._parser.add_argument(
+            "--kaggle-api-key",
+            type=str,
+            required=False,
+            help="Kaggle API key. Needed for gated models like Llama2. You can find your API key at https://kaggle.com/settings.",
+        )
 
     def _download_cmd(self, args: argparse.Namespace) -> None:
+        if args.source == "huggingface":
+            return self._download_from_huggingface(args)
+        if args.source == "kaggle":
+            return self._download_from_kaggle(args)
+
+        msg = f'Unsupported model source {args.source}. Supported values are "huggingface" (default) or "kaggle".'
```

Comment:
I don't think it will ever be able to hit this section of the code b/c the ArgumentParser "choice" functionality blocks anything other than huggingface or kaggle.

Reply:
this is true. I think, in the moment, I saw that more as a side-effect of `ArgumentParser` that wasn't super clear in the context of this function. I can remove it!